src/main/java/seedu/duke/commands/Add.java

```diff
@@ -0,0 +1,125 @@
+package seedu.duke.commands;
+
+import seedu.duke.Module;
+import seedu.duke.ModuleList;
+import seedu.duke.exceptions.InvalidInputContentException;
+import seedu.duke.exceptions.InvalidInputFormatException;
+
+public class Add extends Command {
+    private Module mod;
+
+    /**
+     * Contructor of Add class to initialize an object of class Add.
+     * @param input the input message to be used to initialize the variables
+     * @throws InvalidInputFormatException exception which is thrown if the format of the input is wrong
+     * @throws InvalidInputContentException exception to be thrown if the input content is empty
+     */
+    public Add(String input) throws InvalidInputFormatException, InvalidInputContentException {
+        checkFormat(input);
+        int[] indexes = positions(input);
+        checkContent(input, indexes);
+        addition(input, indexes);
+    }
+
+    /**
+     * Function to extract details from input text and create a new module with the details extracted.
+     * Extracted details include course, semester, mc, and grade.
+     * @param input the input entered by the user
+     * @param indexes an array containing the positions from which the details need to be extracted
+     */
+    private void addition(String input, int[] indexes) {
+        String course = extractingContent(input, indexes[0], indexes[1]);
+        String semester = extractingContent(input, indexes[2], indexes[3]);
+        String mcString = extractingContent(input, indexes[4], indexes[5]);
+        int mc = Integer.parseInt(mcString);
+        String grade = extractingContent(input, indexes[6], indexes[7]);
+
+        this.mod = new Module(course.toUpperCase(), semester.toUpperCase(), grade.toUpperCase(), mc);
```

Comment:
This may be a complicated expression. Consider moving the changing of cases into the constructor.